pkg/status/combinedstatus-resolution.go

```diff
@@ -0,0 +1,308 @@
+func (c *combinedStatusResolution) evaluateWorkStatusAgainstStatusCollector(celEvaluator *celEvaluator,
+	workStatusIdentifier util.ObjectIdentifier, workStatusContent *runtime.RawExtension,
+	scData *statusCollectorData) (bool, error) {
+	wsData, ok := scData.workStatusRefToData[workStatusIdentifier]
+	if !ok {
+		wsData = workStatusData{}
+		scData.workStatusRefToData[workStatusIdentifier] = wsData
+	}
+
+	updated := false
+	// evaluate all clauses
+	if scData.filter != nil {
+		eval, err := celEvaluator.Evaluate(string(*scData.filter), workStatusContent)
+		if err != nil {
+			return false, fmt.Errorf("failed to evaluate filter expression: %w", err)
+		}
+
+		if eval.Type().TypeName() != "bool" {
+			return false, fmt.Errorf("filter expression must evaluate to a boolean")
+		}
+
+		fiterEval := eval.Value().(bool)
+		updated = fiterEval != wsData.filterEval
+		wsData.filterEval = fiterEval
+	}
+
+	// evaluate select
+	selectEvals := make(map[string]any)
+	for _, selectNamedExp := range scData.selectClause {
+		eval, err := celEvaluator.Evaluate(string(selectNamedExp.Def), workStatusContent)
+		if err != nil {
+			return false, fmt.Errorf("failed to evaluate select expression: %w", err)
+		}
+
+		selectEval := eval.Value()
+		updated = updated || selectEval != wsData.selectEval[selectNamedExp.Name]
+		selectEvals[selectNamedExp.Name] = selectEval
+	}
+
+	// evaluate groupBy
+	groupByEvals := make(map[string]any)
+	for _, groupByNamedExp := range scData.groupBy {
+		eval, err := celEvaluator.Evaluate(string(groupByNamedExp.Def), workStatusContent)
+		if err != nil {
+			return false, fmt.Errorf("failed to evaluate groupBy expression: %w", err)
+		}
+
+		groupByEval := eval.Value()
+		updated = updated || groupByEval != wsData.groupByEval[groupByNamedExp.Name]
+		groupByEvals[groupByNamedExp.Name] = groupByEval
+	}
+
+	// evaluate combinedFields
+	combinedFieldEvals := make(map[string]any)
+	for _, combinedFieldNamedAgg := range scData.combinedFields {
+		if combinedFieldNamedAgg.Type == v1alpha1.AggregatorTypeCount {
+			// count does not require a subject - mark the evaluation with the type
+			updated = updated || wsData.combinedFieldsEval[combinedFieldNamedAgg.Name] != v1alpha1.AggregatorTypeCount
+			combinedFieldEvals[combinedFieldNamedAgg.Name] = v1alpha1.AggregatorTypeCount
+			continue
+		}
+
+		// evaluate subject which should not be nil since the statuscollector is valid
+		eval, err := celEvaluator.Evaluate(string(*combinedFieldNamedAgg.Subject), workStatusContent)
+		if err != nil {
+			return false, fmt.Errorf("failed to evaluate combinedFields expression: %w", err)
+		}
+
+		combinedFieldEval := eval.Value()
+		updated = updated || combinedFieldEval != wsData.combinedFieldsEval[combinedFieldNamedAgg.Name]
+		combinedFieldEvals[combinedFieldNamedAgg.Name] = combinedFieldEval
+	}
+
+	// update the workstatus data
+	wsData.selectEval = selectEvals
+	wsData.groupByEval = groupByEvals
+	wsData.combinedFieldsEval = combinedFieldEvals
```

Comment:
`wsData` is not a pointer.

Reply:
.. so these asignments to a local struct that is not returned have no meaningful effect.

Reply:
Are you referring to the maintenance of CombinedStatus objects? Such as when changed externally or deleted for whatever reason? This will be handled when the TODO stated in the resolver (`// TODO: API for generating/comparing combinedstatus resolutions`) is addressed and used.